test/check/check.go

```diff
@@ -0,0 +1,270 @@
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	securityapi "github.com/openshift/origin/pkg/security/apis/security"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"gopkg.in/yaml.v2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/names"
+	"github.com/kubevirt/cluster-network-addons-operator/test/log"
+)
+
+func Assert(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func CheckComponentsDeployment(components []Component) error {
+	for _, component := range components {
+		if err := checkForComponent(&component); err != nil {
+			log.Fail("Component %q has not created", component.ComponentName)
+			return err
+		}
+		log.Ok("Component %q is available", component.ComponentName)
+	}
+	return nil
+}
+
+func CheckComponentsRemoval(components []Component) error {
+	for _, component := range components {
+		// TODO: Once finalizers are implemented, we should switch to using
+		// once-time checks, since after NodeNetworkState removal, no components
+		// should be left over.
+		if err := waitForComponentRemoval(&component); err != nil {
+			log.Fail("Component %q has not been removed within given timeout", component.ComponentName)
+			return err
+		}
+		log.Ok("Component %q is gone", component.ComponentName)
+	}
+	return nil
+}
+
+func CheckConfigCondition(conditionType ConditionType, conditionStatus ConditionStatus, timeout time.Duration) error {
+	if err := waitForCondition(opv1alpha1.NetworkAddonsConditionType(conditionType), corev1.ConditionStatus(conditionStatus), timeout); err != nil {
+		log.Fail("Condition %q has not reached desired state %q within given timeout", conditionType, conditionStatus)
+		return err
+	}
+	log.Ok("Condition %q has reached desired state %q", conditionType, conditionStatus)
+	return nil
+}
+
+func checkForComponent(component *Component) error {
```

Comment:
It's possible to pass a function here and the function declaration to be runtime.Object so we remove the if statement.

Another approach would be the typical Runnable interface and pass it to this function.

Reply:
I don't understand your comment. Could you give some examples?

Reply:
Na forget about this, was a total brain fart from my part.